Adding VASP Application Edits for Gila (#847)
NREL/HPC: docs/Documentation/Applications/vasp.md

    mpirun -npernode 4 vasp_std &> out
    ```

## VASP on Gila

#### CPU
??? example "Sample job script: Gila - VASP 6.4.1 CPU (Intel - Intel MPI)"
  
    ```
    #!/bin/bash
    #SBATCH --job-name=vasp
    #SBATCH --nodes=1
    #SBATCH --ntasks=60

>     #SABTCH --mem=60G

Reviewer: ```suggestion
    #SBATCH --mem=60G
```

    #SBATCH --time=8:00:00
    #SBATCH --error=std.err
    #SBATCH --output=std.out
    #SBATCH --partition=amd
    #SBATCH --account=myaccount

    ml application
    ml vasp-intel/6.4.1

    # some extra lines that have been shown to improve VASP performance on Gila when compiled against an Intel toolchain
    ulimit -s unlimited
    export I_MPI_ADJUST_REDUCE=3

    srun -n 60 vasp_std

    # If the multi-node calculations are breaking, replace the srun line with this line
    
    ```
??? example "Sample job script: Gila - VASP 6.4.1 CPU (GCC - OpenMPI)"
  
    ```
    #!/bin/bash
    #SBATCH --job-name=vasp
    #SBATCH --nodes=1
    #SBATCH --ntasks=60
    #SBATCH --mem=60G
    #SBATCH --time=8:00:00
    #SBATCH --error=std.err
    #SBATCH --output=std.out
    #SBATCH --partition=amd
    #SBATCH --account=myaccount

>     ml vasp-gcc/6.4.1

Reviewer: Does the application module need to be loaded?

    ### Using mpirun as opposed to srun due to OpenMPI 5.0 compatibility issues with Slurm Process Manager
    mpirun -np 60 vasp_std
    ```

#### GPU

??? example "Sample job script: Gila - VASP 6.4.1 GPU (A100) (OpenACC)"
  
    ```
    #!/bin/bash
    #SBATCH --job-name=vasp
    #SBATCH --nodes=1
    #SBATCH --time=8:00:00
    #SBATCH --ntasks=1
    #SBATCH --mem=60G
    #SBATCH --gpus=1
    #SBATCH --error=std.err
    #SBATCH --output=std.out
    #SBATCH --partition=gpu-intel-a100-80g
    #SBATCH --account=myaccount
    
    ml application
    ml vasp-gpu-a100/6.4.1
    export CUDA_VISIBLE_DEVICES=0
    export OMPI_MCA_opal_cuda_support=1

    ### Using mpirun as opposed to srun due to OpenMPI 5.0 compatibility issues with Slurm Process Manager
    mpirun -np 1 60 vasp_std

    ```

> ??? example "Sample job script: Gila - VASP 6.4.1 GPU (Grace-Hopper) (OpenACC)"

Reviewer: ```suggestion
??? example "Sample job script: Gila - VASP 6.4.1 GPU (Grace Hopper) (OpenACC)"
```

Reviewer: Using naming standard "Grace Hopper", so this needs to be fixed throughout the page

  
    ```
    #!/bin/bash
    #SBATCH --job-name=vasp
    #SBATCH --nodes=1
    #SBATCH --time=8:00:00
    #SBATCH --ntasks=1
    #SBATCH --mem=60G
    #SBATCH --gpus=1
    ##SBATCH --error=std.err
    ##SBATCH --output=std.out
    #SBATCH --partition=gh
    #SBATCH --account=myaccount

    export OMPI_MCA_opal_cuda_support=1
    export CUDA_VISIBLE_DEVICES=0
    ml application
    ml vasp-gpu-grace

    ### Using mpirun as opposed to srun due to OpenMPI 5.0 compatibility issues with Slurm Process Manager
    mpirun -np 1 60 vasp_std

    ```

??? note "Performance Notes"

>     On Gila, VASP runs more performantly on a single node. If many cores are needed for your VASP calculation, it is recommended to run VASP on a singe node in the amd partition (60 cores/node), which provides the largest numbers of cores per node and use the following settings that have been shown to work well for multi-node jobs on 2 nodes. Intel-MPI is required to run multi-node VASP jobs on Gila, where as OpenMPI can only be run on single node VASP jobs.

Reviewer: ```suggestion
    On Gila, VASP runs more performantly on a single node. If many cores are needed for your VASP calculation, it is recommended to use a single node in the `amd` partition (60 cores/node), which provides the largest number of cores per node and use the following settings that have been shown to work well for multi-node jobs on 2 nodes. Intel-MPI is required to run multi-node VASP jobs on Gila, whereas OpenMPI can only be run on single node VASP jobs.
```

Author: Changes made!

>     On Gila, VASP runs more performantly on a single node. If many cores are needed for your VASP calculation, it is recommended to run VASP on a singe node in the amd partition (60 cores/node), which provides the largest numbers of cores per node and use the following settings that have been shown to work well for multi-node jobs on 2 nodes. Intel-MPI is required to run multi-node VASP jobs on Gila, where as OpenMPI can only be run on single node VASP jobs.

Reviewer: "and use the following setting" 

Which settings are you referring to?

>     Between the two GPU VASP versions available, the Grace Hopper version is more performant than the NVIDIA A100 version on a single GPU, but each NVIDIA A100 node has more GPUs than the Grace Hopper nodes. However, neither GPU version can be run multi-node - the underlying compilation architecture of both (nvhpc) relies on its own implementation of OpenMPI 5.0, where we run into the same process manager issues in Slurm as we do on the AMD nodes. This issue is a work in progress.

Reviewer: ```suggestion
    Between the two GPU VASP versions available, the Grace Hopper version is more performant than the NVIDIA A100 version on a single GPU, but each NVIDIA A100 node has more GPUs than the Grace Hopper nodes. However, neither GPU version can be run multi-node. The underlying compilation architecture of both (nvhpc) relies on its own implementation of OpenMPI 5.0, where we run into the same process manager issues in Slurm as we do on the AMD nodes. This issue is a work in progress.
```

>     Between the two GPU VASP versions available, the Grace-Hopper version is more performant than the NVIDIA A100 version on a single GPU, but each NVIDIA A100 node has more GPUs than the Grace-Hopper nodes. However, both GPU versions can not be run multinode - the underlying compilation architecture of both (nvhpc) relies on its own implementation of OpenMPI 5.0, where we run into the same process manager issues in Slurm as we do on the AMD nodes - this issue is a work in progress.

Reviewer: ```suggestion
    Between the two GPU VASP versions available, the Grace-Hopper version is more performant than the NVIDIA A100 version on a single GPU, but each NVIDIA A100 node has more GPUs than the Grace-Hopper nodes. However, neither GPU version can be run multi-node - the underlying compilation architecture of both (nvhpc) relies on its own implementation of OpenMPI 5.0, where we run into the same process manager issues in Slurm as we do on the AMD nodes. This issue is a work in progress.
```

>     Between all VASP versions running on a single node, the Grace-Hopper version is the most performant. VASP multi-node jobs can only be run with Intel-MPI, and the performance in that can vary according to the size of the given system. Experimentation may be necessary to run the most optimal multi-node jobs using Intel-MPI.

Reviewer: ```suggestion
    Between all VASP versions running on a single node, the Grace-Hopper version is the most performant. VASP multi-node jobs can only be run with Intel-MPI, and the performance can vary according to the size of the given system. Experimentation may be necessary to run the most optimal multi-node jobs using Intel-MPI.
```

Author: Changes made!
